var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -50,12 +63,81 @@ class Mesa(AutotoolsPackage):
     depends_on('libxext')
     depends_on('libxdamage')
     depends_on('libxfixes')
+    depends_on('expat')
 
     depends_on('glproto@1.4.14:', type='build')
     depends_on('dri2proto@2.6:', type='build')
     depends_on('dri3proto@1.0:', type='build')
     depends_on('presentproto@1.0:', type='build')
     depends_on('pkg-config@0.9.0:', type='build')
+    depends_on('binutils', type='build')
 
     # TODO: Add package for systemd, provides libudev
     # Using the system package manager to install systemd didn't work for me
+
+    def url_for_version(self, version):
+        """Handle mesa version-based custom URLs."""
+        latest_mver = '11'
+
+        if version < Version(latest_mver):
+            base = Mesa.base_url + "/older-versions/%s.x" % (version.up_to(1))
+        else:
+            base = Mesa.base_url 
+        if version > Version('10.4'):
+            name = 'mesa'
+        else:
+            name = 'MesaLib'
+        ret = base + "/%s/%s-%s.tar.gz" % (version, name, version)
+        return ret
+
+    def check_variants(self, var='', ver='', ref=''):
+        if var and ver:
+            error = "to be safe avoid '{variant}' with version '{version}' see:\n {reference}" # noqa
+            if var in self.spec and self.spec.satisfies(ver):
+                raise RuntimeError(
+                    error.format(variant=var, version=ver, reference=ref))
+
+#    def install(self, spec, prefix):
+#        configure('--prefix={0}'.format(prefix))
+#        self.check_variants(
+#            var='+gallium',
+#            ver='@:9.2.2',
+#            ref='http://www.paraview.org/Wiki/ParaView/ParaView_And_Mesa_3D')
+
+    def configure_args(self):
+        self.check_variants(
```

Comment:
the link here is a general description, one can just keep it as a comment above
```
# see http://www.paraview.org/Wiki/ParaView/ParaView_And_Mesa_3D
conflicts('+galium', when='@:9.2.2')
```
or we need to ask @alalazo to allow custom messages in `confligts()` statement :smile: